src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -24,6 +24,8 @@
  */
 final class ComponentRenderer implements ComponentRendererInterface
 {
+    private array $templateClasses = [];
```

Comment:
We should reset this state I think.

Reply:
Done, and added some (not so) minor optim in Factory again :|